src/collaboration-platform/GitHubCollaborationPlatform.ts

```diff
@@ -5,9 +5,33 @@ import type {
   PullRequestInfo,
 } from "../shared/types";
 import type { CollaborationPlatform } from "./CollaborationPlatform";
+import { getOctokit } from "./OctokitAuth";
+import nodegit from "nodegit";
+
+async function repoPathToOwnerAndRepo(
+  repoPath: string,
+): Promise<{ owner: string; repo: string }> {
+  const repoResult = await nodegit.Repository.open(repoPath);
+  // TODO: Handle remotes that are not named "origin"
+  const remoteResult = await repoResult.getRemote("origin");
+  const remoteUrl = remoteResult.url();
+  // Sample URLs:
+  // "https://github.com/taneliang/stack-attack"
+  // "git@github.com:taneliang/stack-attack.git"
+  // "git@github.com:taneliang/hello.world"
+  const [rawRepo, owner] = remoteUrl
+    .split("/")
+    .flatMap((s) => s.split(":"))
+    .reverse();
+  const repo = rawRepo.endsWith(".git")
+    ? rawRepo.substr(0, rawRepo.length - 4)
+    : rawRepo;
+  return { owner, repo };
+}
 
 export class GitHubCollaborationPlatform implements CollaborationPlatform {
   private repoPath: string;
```

Comment:
Ah, looks like `repoPath` doesn't give us the information this class needs. We shouldn't be touching Git (or Nodegit) at all in this class. I think the best solution is to change `GitHubCollaborationPlatform` to store `owner` and `repo` instead, and have the `constructStacker` factory function compute the owner and repo and pass it in.

```suggestion
  private owner: string;
  private repo: string;
```